armi/reactor/converters/tests/test_axialExpansionChanger.py

```diff
@@ -563,6 +563,50 @@ def test_isFuelLocked(self):
             self.assertEqual(the_exception.error_code, 3)
 
 
+class TestSpecifyTargetComponent(unittest.TestCase):
+    """verify specifyTargetComponent method is properly updating _componentDeterminesBlockHeight"""
+
+    def setUp(self):
+        Base.setUp(self)
+        self.a = buildTestAssemblyWithFakeMaterial(name="FakeMatException")
+        self.obj.setAssembly(self.a)
+
+    def test_specifyTargetComponent(self):
+        initialComponentDeterminesBlockHeight = (
+            self.obj.expansionData._componentDeterminesBlockHeight.copy()
+        )
+        # build a test block
+        b = HexBlock("fuel", height=10.0)
+        fuelDims = {"Tinput": 25.0, "Thot": 25.0, "od": 0.76, "id": 0.00, "mult": 127.0}
+        cladDims = {"Tinput": 25.0, "Thot": 25.0, "od": 0.80, "id": 0.77, "mult": 127.0}
+        fuel = Circle("fuel", "FakeMat", **fuelDims)
+        clad = Circle("clad", "FakeMat", **cladDims)
+        b.add(fuel)
+        b.add(clad)
+        # call method, and check that _componentDeterminesBlockHeight has been updated
+        self.obj.expansionData.specifyTargetComponent(b)
+        newEntry = dict(
+            set(self.obj.expansionData._componentDeterminesBlockHeight.items())
+            - set(initialComponentDeterminesBlockHeight.items())
+        )
+        self.assertEqual(newEntry, {fuel: True})
```

Comment:
@john-science can you check that this is an OK way to test for the update to the dictionary? Also note line 575 above, where I made a copy to snapshot the dict prior to updating.